tests/c/sdiv.newcg.c

```diff
@@ -0,0 +1,77 @@
+// ignore-if: test $YK_JIT_COMPILER != "yk" -o "$YKB_TRACER" = "swt"
+// Run-time:
+//   env-var: YKD_LOG_IR=-:jit-pre-opt
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_LOG_JITSTATE=-
+//   stderr:
+//     jitstate: start-tracing
+//     sdiv 10922
+//     sdiv2 715827882
+//     sdiv3 1431655764
+//     sdiv4 *
+//     jitstate: stop-tracing
+//     --- Begin jit-pre-opt ---
+//     ...
+//     %{{1}}: i16 = SDiv %{{2}}, 3i16
+//     ...
+//     %{{3}}: i32 = SDiv %{{4}}, 3i32
+//     ...
+//     %{{5}}: i64 = SDiv %{{6}}, 3i64
+//     ...
+//     %{{7}}: i8 = SDiv %{{8}}, 3i8
+//     ...
+//     --- End jit-pre-opt ---
+//     sdiv 10922
+//     sdiv2 715827882
+//     sdiv3 1431655764
+//     sdiv4 *
+//     jitstate: enter-jit-code
+//     sdiv 10922
+//     sdiv2 715827882
+//     sdiv3 1431655764
+//     sdiv4 *
+//     sdiv 10922
+//     sdiv2 715827882
+//     sdiv3 1431655764
+//     sdiv4 *
+//     jitstate: deoptimise
+//     exit
+
+// Test signed division.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc = yk_location_new();
+
+  int i = 4;
+  short int num1 = 32767;
+  int num2 = 2147483647;
+  long long num3 = 4294967294;
+  signed char num4 = 127;
+  NOOPT_VAL(loc);
+  NOOPT_VAL(i);
+  while (i > 0) {
+    yk_mt_control_point(mt, &loc);
+    short int sdiv = num1 / 3;
```

Comment:
Since `sdiv` and `num1` (and some other variables) aren't `NOOPT_VAL`ed, I think it's possible that at `-O2` these will be constant folded. It's probably best to `NOOPT_VAL` the relevant variables.

Reply:
Should be enough if I `NOOPT_VAL` the `numX` variables. Then the `sdivX` variables can't be optimised away anymore.

Reply:
[8a16127](https://github.com/ykjit/yk/pull/1174/commits/8a16127bd424917c050246ae854066ea5fcae1e9)